src/commands/app/install.js

```diff
@@ -139,6 +143,67 @@ class InstallCommand extends BaseCommand {
       }
     })
   }
+
+  /**
+   * An annotation called code-download will be added to all actions in app.config.yaml
+   * (and linked yaml configs for example in extensions). This value will be set to false.
+   * The annotation will by default be true if not set.
+   *
+   * @param {object} outputPath the path to the app package
+   * @param {object} appConfig the app's configuration file
+   */
+  async addCodeDownloadAnnotation (outputPath, appConfig) {
+    this.spinner.start('Adding code-download annotations...')
+    // get each annotation key relative to the file it is defined in
+    /// iterate only over extensions that have actions defined
+    const fileToAnnotationKey = {}
+    Object.entries(appConfig.all)
+      .filter(([_, extConf]) => extConf.manifest?.full?.packages)
+      .forEach(([ext, extConf]) => {
+        Object.entries(extConf.manifest.full.packages)
+          .filter(([pkg, pkgConf]) => pkgConf.actions)
+          .forEach(([pkg, pkgConf]) => {
+            Object.entries(pkgConf.actions).forEach(([action, actionConf]) => {
+              const baseFullKey = ext === 'application'
+                ? `application.runtimeManifest.packages.${pkg}.actions.${action}`
+                : `extensions.${ext}.runtimeManifest.packages.${pkg}.actions.${action}`
+
+              let index
+              if (actionConf.annotations) {
+                index = appConfig.includeIndex[`${baseFullKey}.annotations`]
+              } else {
+                // the annotation object is not defined, take the parent key
+                index = appConfig.includeIndex[baseFullKey]
+              }
+              if (!fileToAnnotationKey[index.file]) {
+                fileToAnnotationKey[index.file] = []
+              }
+              fileToAnnotationKey[index.file].push(index.key) // index.key is relative to the file
+            })
+          })
+      })
+
+    // rewrite config files
+    for (const [file, keys] of Object.entries(fileToAnnotationKey)) {
+      const configFilePath = path.join(outputPath, file)
+      const { values } = loadConfigFile(configFilePath)
+
+      keys.forEach(key => {
+        const object = getObjectValue(values, key)
+        if (key.endsWith('.annotations') || key === 'annotations') {
+          // object is the annotations object
+          object['code-download'] = false
```

Comment:
Did we get confirmation from the Runtime team that the annotation name is `code-download` ?

Reply:
Per @asteed: 

> It’s going to be `disable-download`. We wanted to adjust to follow the syntax for other reserved annotations

Reply:
Just pushed a new commit with this change